Algorithms.Tests/Search/ExponentialSearcherTests.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Algorithms.Search;
+using FluentAssertions;
+using NUnit.Framework;
+using NUnit.Framework.Internal;
+
+namespace Algorithms.Tests.Search
+{
+    public static class ExponentialSearcherTests
+    {
+        [Test]
+        public static void FindIndex_ItemPresent_IndexCorrect([Random(1, 1000, 100)] int n)
+        {
+            // Arrange
+            var subject = new ExponentialSearcher();
+            var random = Randomizer.CreateRandomizer();
+            var arrayToSearch = Enumerable.Range(0, n).Select(_ => random.Next(0, 1000)).OrderBy(x => x).ToArray();
+            var selectedIndex = random.Next(0, n);
+
+            // Act
+            var actualIndex = searcher.FindIndex(arrayToSearch, arrayToSearch[selectedIndex]);
+
+            // Assert
+            Assert.AreEqual(arrayToSearch[selectedIndex], arrayToSearch[actualIndex]);
+        }
+        [Test]
+        public static void FindIndex_ItemMissing_MinusOneReturned(
+            [Random(0, 1000, 10)] int n,
+            [Random(-100, 1100, 10)] int missingItem)
+        {
+            // Arrange
+            var subject = new ExponentialSearcherr<int>();
+            var random = Randomizer.CreateRandomizer();
+            var collection = Enumerable.Range(0, n)
+                .Select(_ => random.Next(0, 1000))
+                .Where(x => x != missingItem)
+                .OrderBy(x => x).ToList();
+
+            // Act
+            var actualIndex = subject.FindIndex(collection, missingItem);
+
+            // Assert
+            Assert.AreEqual(-1, actualIndex);
+        }
+
+        [Test]
+        public static void FindIndex_ArrayEmpty_MinusOneReturned([Random(100)] int itemToSearch)
+        {
+            // Arrange
+            var searcher = new ExponentialSearcher<int>();
+            var arrayToSearch = new int[0];
+
+            // Act
+            var actualIndex = searcher.FindIndex(arrayToSearch, itemToSearch);
+
+            // Assert
+            Assert.AreEqual(-1, actualIndex);
```

Comment:
Please use FluentAssertions here and above

Reply:
I added FluentAssertions and addressed the other feedback, thanks!